base/missing.jl

```diff
@@ -259,6 +259,14 @@ keys(itr::SkipMissing) =
     v
 end
 
+"""
+    hasmissing(itr)
+
+Checks if itr has missing values.
+"""
+hasmissing(x::Base.SkipMissing) = false
+hasmissing(itr) = ~(collect(skipmissing(itr)) == collect(itr))
```

Comment:
```suggestion
hasmissing(itr) = collect(skipmissing(itr)) != collect(itr)
```
This would be the more idiomatic way to write this. However, there are several issues with this: it collects the iterable twice, which doesn't work for many iterables and it's really inefficient.

Reply:
Wouldn't it be better to just do `missing in itr`?

Reply:
Good point about iterables not supporting double collection. But I wasn't sure how else to support both iterables and concrete collections.